src/generator/cpp/index.ts

```diff
@@ -243,9 +277,41 @@ export class CPPGenerator implements CodeGenerator {
     targetExpr: string,
   ): string {
     const assign = (val: string) => `  ${targetExpr} = ${val};`;
+
+    if (field.isArray) {
+      let decoder: string;
+      let cppType: string;
```

Comment:
Don't use let for `decoder` and `cppType`, make private methods, call them, and assign to const if needed:

`const cppType = this.getCppType(params)`